src/components/roar-page/useHeader.tsx

```diff
@@ -43,14 +60,150 @@ const RoarLogo = (): JSX.Element => {
+    const ref = React.useRef()
+    refsToTrack.push(ref as any)
+    return ref as any
+  }
+
+  const internalSectionRefs: SectionRefs<Section | 'hero'> = internalSections.reduce(
+    (refs, section) => ({
+      ...refs,
+      [section]: React.useRef<HTMLElement>(),
+    }),
+    {
+      hero: React.useRef<any>()
+    } as any
+  )
+
+  const internalLinkRefs: SectionRefs<Section | 'hero'> = internalSections.reduce(
+    (refs, section) => ({
+      ...refs,
+      [section]: React.useRef<HTMLElement>(),
+    }),
+    {
+      hero: React.useRef<any>()
+    } as any
+  )
+
+  const headerRef = React.useRef<HTMLDivElement>()
+
+  React.useEffect(
+    () => {
+      const headerElement = headerRef.current!
+
+      let headerIsFixed = getComputedStyle(headerElement).position === 'fixed' // tslint:disable-line:no-let
+
+      function onScroll(): void {
+        if (!headerIsFixed) return
+
+        const headerBottom = headerElement.getBoundingClientRect().bottom
+
+        refsToTrack.forEach(ref => {
+          if (!ref.current) return
+          const rect = ref.current.getBoundingClientRect()
+          if (!rect.height && !rect.width) return
+          const elementTop = rect.top
+          const elementDistance = elementTop - headerBottom
+          const elementOpacity = Math.max(0, Math.min(1, (elementDistance + 10) / 100))
+          ref.current.style.opacity = String(elementOpacity)
+        })
+
+        const screenMidpoint = screen.availHeight / 2
+
+        const currentSection = Object.keys(internalSectionRefs).find((section: Section) => {
+          const ref = internalSectionRefs[section]
+          const { top, bottom } = ref.current!.getBoundingClientRect()
+          return top <= screenMidpoint && bottom >= screenMidpoint
+        })!
+
+        // If there is no current section, leave whatever link is already active as active
+        if (currentSection) {
+          forEach(internalLinkRefs, (ref, key) => {
+            if (key === currentSection) {
+              ref.current!.classList.add('active')
+            } else {
+              ref.current!.classList.remove('active')
+            }
+          })
+        }
+      }
+
+      function onResize(): void {
+        headerIsFixed = getComputedStyle(headerElement).position === 'fixed'
+        onScroll()
+      }
+
+      onScroll()
+      addEventListener('scroll', onScroll, { passive: true })
+      addEventListener('resize', onResize)
+
+      return () => {
+        removeEventListener('scroll', onScroll)
+        removeEventListener('resize', onResize)
+      }
+    },
+    refsToTrack.map(ref => ref.current)
+  )
+
+  React.useEffect(() => {
+    const hash = location.hash.slice(1)
+    if (hash && internalSections.includes(hash as any)) {
+      // Not sure why this requires a timeout, but it seems to
+      setTimeout(() => internalSectionRefs[hash as Section].current.scrollIntoView())
+    }
+  }, [location.hash])
```

Comment:
The links already include a `scrollIntoView`. Why do we need this scroll what the hash changes?

Reply:
In case we want to link to `/roar#learn-more` although that function wasn't working. It's now fixed to use kebabCase